swift/ql/test/query-tests/Security/CWE-311/testGRDB.swift

```diff
@@ -0,0 +1,214 @@
+    fetchableRecord.fetchAll(stmt, arguments: [password]) // BAD
+    fetchableRecord.fetchAll(stmt, sql: "", arguments: [harmless]) // GOOD
+    fetchableRecord.fetchAll(stmt, arguments: [harmless]) // GOOD
+    fetchableRecord.fetchAll(stmt, sql: "", arguments: [password], adapter: nil) // BAD
+    fetchableRecord.fetchAll(stmt, arguments: [password], adapter: nil) // BAD
+    fetchableRecord.fetchAll(stmt, sql: "", arguments: [harmless], adapter: nil) // GOOD
+    fetchableRecord.fetchAll(stmt, arguments: [harmless], adapter: nil) // GOOD
+
+    fetchableRecord.fetchSet(stmt, sql: "", arguments: [password]) // BAD
+    fetchableRecord.fetchSet(stmt, arguments: [password]) // BAD
+    fetchableRecord.fetchSet(stmt, sql: "", arguments: [harmless]) // GOOD
+    fetchableRecord.fetchSet(stmt, arguments: [harmless]) // GOOD
+    fetchableRecord.fetchSet(stmt, sql: "", arguments: [password], adapter: nil) // BAD
+    fetchableRecord.fetchSet(stmt, arguments: [password], adapter: nil) // BAD
+    fetchableRecord.fetchSet(stmt, sql: "", arguments: [harmless], adapter: nil) // GOOD
+    fetchableRecord.fetchSet(stmt, arguments: [harmless], adapter: nil) // GOOD
+
+    fetchableRecord.fetchOne(stmt, sql: "", arguments: [password]) // BAD
+    fetchableRecord.fetchOne(stmt, arguments: [password]) // BAD
+    fetchableRecord.fetchOne(stmt, sql: "", arguments: [harmless]) // GOOD
+    fetchableRecord.fetchOne(stmt, arguments: [harmless]) // GOOD
+    fetchableRecord.fetchOne(stmt, sql: "", arguments: [password], adapter: nil) // BAD
+    fetchableRecord.fetchOne(stmt, arguments: [password], adapter: nil) // BAD
+    fetchableRecord.fetchOne(stmt, sql: "", arguments: [harmless], adapter: nil) // GOOD
+    fetchableRecord.fetchOne(stmt, arguments: [harmless], adapter: nil) // GOOD
+}
+
+func test(stmt: Statement, password: String, harmless: String) {
+    stmt.execute(arguments: [password]) // BAD
+    stmt.execute(arguments: [harmless]) // GOOD
+
+    stmt.setArguments([password]) // BAD
+    stmt.setArguments([harmless]) // GOOD
+}
+
+func testCommonTableExpression(password: String, harmless: String) {
+    let _ = CommonTableExpression(named: "", sql: "", arguments: [password]) // BAD
+    let _ = CommonTableExpression(named: "", sql: "", arguments: [harmless]) // GOOD
+    let _ = CommonTableExpression(named: "", columns: nil, sql: "", arguments: [password]) // BAD
+    let _ = CommonTableExpression(named: "", columns: nil, sql: "", arguments: [harmless]) // GOOD
+    let _ = CommonTableExpression(recursive: false, named: "", sql: "", arguments: [password]) // BAD
+    let _ = CommonTableExpression(recursive: false, named: "", sql: "", arguments: [harmless]) // GOOD
+    let _ = CommonTableExpression(recursive: false, named: "", columns: nil, sql: "", arguments: [password]) // BAD
+    let _ = CommonTableExpression(recursive: false, named: "", columns: nil, sql: "", arguments: [harmless]) // GOOD
+}
```

Comment:
There's another way for sensitive data to end up in the database via GRDB - as part of a database record, without using SQL.  I think the pattern we'd be looking for is something like:
```
struct MyData: MutablePersistableRecord { // or PersistableRecord
	var data: String
}

...

var myData = MyData(
  data: password
)

OR

myData.data = password
```
This might be best left for another PR.

Reply:
> There's another way for sensitive data to end up in the database via GRDB - as part of a database record, without using SQL. 

Good point! And that will probably be the more common case too.

> This might be best left for another PR.

I think so too, this one has enough changes already.

Reply:
I've added a note to the issue.